content/blog/oauth-oidc.md

```diff
@@ -0,0 +1,88 @@
++++
+title = "OAuth 2.0, OIDC and Biscuit"
+description = "How Biscuit can fit in existing systems"
+date = 2023-06-17T00:09:00+02:00
+draft = false
+template = "blog/page.html"
+
+[taxonomies]
+authors = ["geal"]
+
+[extra]
+lead = "How OAuth 2.0 and OIDC work, and how to integrate Biscuit with them"
++++
+
+##OAuth 2
+
+[OAuth 2.0](https://oauth.net/2/) is a serie of specifications describing authorization and delegation flows between services. It looks at service access through a list of roles:
+* *resource owner*: an agent that can give access to a resource. That can be a human, but also a service
+* *resource server*: service holding the resources (data, specific actions to launch, etc), requiring authorization from the resource owner to give access to a client
+* *client*: an application that accesses the resources on the resource server, on behalf on the resource owner. It could be anything, anywhere, like a mobile client in your phone that accesses data when you want to consult it, or a remote service that will periodically access your data to back it up
+* *authorization server*: a service that issues access tokens used by the client to tell the resource server that they represent the resource owner
+* *access token*: this can be anything, like a random string, or a cryptographic token like Biscuit, as long as the resource server can use it to recognize a specific resource owner
+
+Depending on the system, some of those roles can be mixed: the resource server could also be the authorization server if they are in the same monomithic service, and the resource server and client could be the same web application.
+Based on those roles, OAuth 2.0 defines a list of authorization flows, and they all follow a general pattern:
+* the resource owner wants the client to access their resources
+* the client contacts the authorization server to start the flow
+* the authorization server requests that the resource owner explicitely accepts it, generally by redirecting the resource owner to a page on the autorization server, eventually requiring username, password, etc
```

Comment:
```suggestion
* the authorization server requests that the resource owner explicitly accepts it, generally by redirecting the resource owner to a page on the autorization server, eventually requiring username, password, etc
```

eventually as in finally? Or potentially/optionally?

Reply:
As in optionally. Some don't even verify that the token works

Reply:
```suggestion
* the authorization server requests that the resource owner explicitely accepts it, generally by redirecting the resource owner to a page on the autorization server, optionally requiring username, password, etc
```

Ok then, eventually means finally, not _éventuellement_.